plugins/linkattacher/src/linkattacherinterface.cpp

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (C) 2013-2015 Fondazione Istituto Italiano di Tecnologia RBCS & iCub Facility & ADVR
+ * Authors: see AUTHORS file.
+ * CopyPolicy: Released under the terms of the LGPLv2.1 or any later version, see LGPL.TXT or LGPL3.TXT
+ */
+
+#include <linkattacherinterface.h>
+
+using namespace gazebo;
+using namespace std;
+using namespace yarp::os;
+
+LinkAttacherInterface::LinkAttacherInterface() : ModelPlugin(), m_network(0), m_rpcport(0)
+{
+}
+
+LinkAttacherInterface::~LinkAttacherInterface()
+{
+  if(m_rpcport)
+  {
+    m_rpcport->close();
+    delete m_rpcport;
+  }
+
+  if(m_network)
+    delete m_network;
+}
+
+void LinkAttacherInterface::Load(physics::ModelPtr _model, sdf::ElementPtr _sdf)
+{
+  if(m_network!=0)
+    return;
+
+  m_network = new yarp::os::Network();
+
+  if(!yarp::os::Network::checkNetwork(GazeboYarpPlugins::yarpNetworkInitializationTimeout))
+  {
+    yError() << LogPrefix << "Load: yarp network does not seem to be available, is the yarpserver running?";
+    return;
+  }
+
+  std::string model_name = _model->GetName();
+
+  yInfo() << LogPrefix << "Load: Model name: " << model_name;
+
+  m_la_server.attachWorldPointer(_model->GetWorld());
+  m_la_server.attachModelPointer(_model);
+
+  bool configuration_loaded = false;
+
+  if( _sdf->HasElement("yarpConfigurationFile") )
+  {
+    std::string ini_file_name = _sdf->Get<std::string>("yarpConfigurationFile");
+    std::string ini_file_path = gazebo::common::SystemPaths::Instance()->FindFileURI(ini_file_name);
+
+    if(ini_file_path != "" && m_parameters.fromConfigFile(ini_file_path.c_str()))
+    {
+      yInfo() << LogPrefix << "Load: Found yarpConfigurationFile: loading from " << ini_file_path;
+      configuration_loaded = true;
+    }
+
+    if(!configuration_loaded)
+    {
+      yError() << LogPrefix << "Load: Failed to load the configuration file";
+      return;
+    }
+
+    std::string portname = m_parameters.find("name").asString();
```

Comment:
What if `"name"` is not present? You should check that it is contained in the `m_parameters` object before accessing it